scipy/special/basic.py

```diff
@@ -2229,24 +2229,37 @@ def perm(N, k, exact=False):
         return vals
 
 
+# http://stackoverflow.com/a/16327037/125507
+def _range_prod(lo, hi):
+    """
+    Product of a range of numbers
+
+    Returns the product of
+    lo * (lo+1) * (lo+2) * ... * (hi-2) * (hi-1) * hi
+    = hi! / (lo-1)!
+    """
+    if lo+1 < hi:
+        mid = (hi + lo) // 2
+        return _range_prod(lo, mid) * _range_prod(mid+1, hi)
+    if lo == hi:
+        return lo
+    return lo*hi
+
+
 def factorial(n, exact=False):
     """The factorial function, n! = special.gamma(n+1).
 
-    If exact is 0, then floating point precision is used, otherwise
+    If exact is False, then floating point precision is used, otherwise
     exact long integer is computed.
 
-    - Array argument accepted only for exact=False case.
-    - If n<0, the return value is 0.
-
     Parameters
     ----------
     n : int or array_like of ints
-        Calculate ``n!``.  Arrays are only supported with `exact` set
-        to False.  If ``n < 0``, the return value is 0.
+        Calculate ``n!``.
     exact : bool, optional
-        The result can be approximated rapidly using the gamma-formula
-        above.  If `exact` is set to True, calculate the
-        answer exactly using integer arithmetic. Default is False.
+        If True, calculate the answer exactly using integer arithmetic.
+        If False, result is approximated rapidly using the gamma-formula
+        above.  Default is False.
```

Comment:
Maybe also modify examples with an array of exact values? It might be worth mentioning that the `exact` mode can return an `object` array